### ISSUE TITLE
Windows support breaks Ansible Collection on Linux

### ISSUE THREAD

**Issue by rpelisse** [ansible-middleware/ansible_collections_jcliff#23]
Step to reproduce:

```
$ ansible-galaxy collection build --force .
Created collection for redhat.jcliff at /home/rpelisse/Repositories/redhat/ansible-jcliff-workspace/ansible_collections/redhat/jcliff/redhat-jcliff-0.0.1.tar.gz
$ ansible-galaxy collection install --force redhat-jcliff-0.0.1.tar.gz 
Process install dependency map
Starting collection install process
Installing 'redhat.jcliff:0.0.1' to '/home/rpelisse/.ansible/collections/ansible_collections/redhat/jcliff'
```

Run a simple playbook:

```
---
- hosts: localhost
  gather_facts: false
  vars:
    ansible_distribution: Fedora
    ansible_os_family: RedHat
    jboss_home: /path/to/jboss-eap-7.2.0/jboss-eap-7.2/
  collections:
    - redhat.jcliff
  roles:
    - jcliff
  tasks:

    - jcliff:
        wfly_home: "{{ jboss_home }}"
        subsystems:
          - system_props:
            - name: jcliff.enabled
              value: 'enabled.plus'
```

Running this playbook will lead Ansible to try to execute the powershell script:

```
PLAY [localhost] *********************************************************************************************************************************************

TASK [jcliff : Collect Supported Operating Systems] **********************************************************************************************************
ok: [localhost] => (item={'key': 'rpm', 'value': ['Fedora', 'CentOS', 'RedHat']})
ok: [localhost] => (item={'key': 'homebrew', 'value': ['MacOSX']})
ok: [localhost] => (item={'key': 'chocolatey', 'value': ['Windows']})

TASK [jcliff : Verify supported Operating Systems] ***********************************************************************************************************
ok: [localhost]

TASK [jcliff : Install JCliff using HomeBrew] ****************************************************************************************************************
skipping: [localhost]

TASK [jcliff : Install JCliff using Chocolatey] **************************************************************************************************************
skipping: [localhost]

TASK [jcliff : Install JCliff using RPM] *********************************************************************************************************************
included: /home/rpelisse/.ansible/collections/ansible_collections/redhat/jcliff/roles/jcliff/tasks/install_rpm.yml for localhost

TASK [jcliff : Add JCliff Yum Repository (Fedora)] ***********************************************************************************************************
ok: [localhost]

TASK [jcliff : Test if package jcliff is already installed] **************************************************************************************************
ok: [localhost]

TASK [jcliff : Install Jcliff using standalone binary] *******************************************************************************************************
skipping: [localhost]

TASK [jcliff] ************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: powershell: command not found\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 127}

PLAY RECAP ***************************************************************************************************************************************************
localhost                  : ok=5    changed=0    unreachable=0    failed=1    skipped=3    rescued=0    ignored=0 

```
This is fixed by deleting the .ps1 scripts from the local install:

`rm ~/.ansible/collections/ansible_collections/redhat/jcliff/plugins/modules/jcliff.ps1`

**Comment by rpelisse** [comment]
@sabre1041 don't forget this issue! This is way more critical than the JMS stuff, right now, the collection is simply broken on Linux...

**Comment by rpelisse** [comment]
This issue seems to be fairly [well explained here](https://www.gitmemory.com/issue/ansible/ansible/61930/531883801), sadly the proposed fix (adding "collections: - ansible.legacy" tot the meta/main.yml to the role) does not appears to work.

**Comment by rpelisse** [comment]
Looking into this [thread](https://groups.google.com/forum/embed/#!topic/ansible-project/zvJFMoN3454), it seems that having powershell in a collection more/less breaks ansible_connection=local. I hope there is a workaround for this...

**Comment by sabre1041** [comment]
@rpelisse cannot reproduce.

Attempted to test on OSX (2.9.0 and 2.9.6) and in a Red Hat container (2.9.9) and ran without issue. 

Used content in master

**Comment by rpelisse** [comment]
@sabre1041 that's annoying... Maybe it is specific to 2.9.10 and/or Python 3.7.7. ?

```
ansible 2.9.10
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/rpelisse/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.7.7 (default, Jun  4 2020, 15:43:14) [GCC 9.3.1 20200408 (Red Hat 9.3.1-2)]
```
I'll try to check this out today.

**Comment by rpelisse** [comment]
I'm definitely have the issue on both my Fedora 32 and RHEL 7.8 running within a Docker container with ansible 2.9.10 (whether or not I run with python2 or python3). 

I've tried retrograding ansible on my Fedora to 2.8.5 and I'm still running into the issue.  I've also try to install the collections on a remote system (another RHEL 7.8) and I also run into the same issue here. I've also try with the ubi8 docker image (so RHEL8) and I got the exact same issue.

As you can't reproduce, I wonder if the issue is not tied to my local set up of Ansible Galaxy, maybe the collections is improperly built? In any case, we have two options here: either we consider this to be a potential issue, and we merge #27 preventively, so that no one can run into this problem, and we chuck it away as fluke. I don't think we should spent more time on this, at least, not unless somebody else can reproduce the problem.

@sabre1041 WDYT?

**Comment by rpelisse** [comment]
Resolved for now, by having a separate feature branch for Windows until we find a satisfying solution.

**Comment by rpelisse** [comment]
As we are running again into this problem, I'm reopen this issue to document the environments where we were able to reproduce:

### RHEL 7.9 with Ansible 2.9.22

OS:

    # cat /etc/redhat-release 
    Red Hat Enterprise Linux Server release 7.9 (Maipo)
    
Ansible:
    
    # ansible-playbook --version
    ansible-playbook 2.9.22
    config file = /etc/ansible/ansible.cfg
    configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
    ansible python module location = /usr/lib/python2.7/site-packages/ansible
    executable location = /usr/bin/ansible-playbook
    python version = 2.7.5 (default, Mar 20 2020, 17:08:22) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]

JCliff Collection version: 

    # grep version /root/.ansible/collections/ansible_collections/wildfly/jcliff/MANIFEST.json 
    "version": "0.0.5", `

Result of local run:

    TASK [jcliff]     ********************************************************************************************************************************************************************************************************************************
    fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: powershell : commande introuvable\n",     "module_stdout":` "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 127}

Same result with a remote run:

    TASK [jcliff] **************************************************************************************************************************
     fatal: [test_server]: FAILED! => {"changed": false, "module_stderr": "Shared connection to test_server closed.\r\n", "module_stdout": "/bin/sh: powershell : commande introuvable\r\n", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 127}

### RHEL 8.4 with Ansible 2.9.22

OS:

     # cat /etc/redhat-release 
     Red Hat Enterprise Linux release 8.4 (Ootpa)

Ansible:

    # ansible --version
    ansible 2.9.22
      config file = /etc/ansible/ansible.cfg
      configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
      ansible python module location = /usr/lib/python3.6/site-packages/ansible
      executable location = /usr/bin/ansible
      python version = 3.6.8 (default, Mar 18 2021, 08:58:41) [GCC 8.4.1 20200928 (Red Hat 8.4.1-1)]


Results of a local run:

    TASK [3trains_wildfly : Fine tune Wildfly configuration] *************************************************************************************************************************************************************************************
    Thursday 01 July 2021  04:22:10 -0400 (0:00:00.389)       0:05:34.672 ********* 
    fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: powershell : commande introuvable\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 127}

**Comment by rpelisse** [comment]
I've tested successfully @hcherukuri changes on both environments. Indeed, the content of [PR55](https://github.com/wildfly-extras/ansible_collections_jcliff/pull/55) does fix the problem on RHEL 7.9 and RHEL 8.4 (whether Ansible is used directly on the host or remotely).
